From c540a04ed40e83b351353b85978b71385f30a756 Mon Sep 17 00:00:00 2001
From: Delphine CC Chiu <Delphine_CC_Chiu@wiwynn.com>
Date: Mon, 25 Sep 2023 11:10:46 +0800
Subject: [PATCH 3/6] Add mctp eids configuration for Yosemite 4

The MCTP specification is defined in DSP0236. As describe in section
8.17.2, we need a configuration for static EIDs. The `mctpd` from
CodeConstuct/mctp will be able to know where those static EIDs located.
The other services will be able to know what those EIDs stand for by the
"Name" as well.

Set the Address property to array so that mctpd could apply both I2C and
I3C address.

schemas: add `EndpointId` with type `string` and `number`.

Tested: check D-Bus path for configuration.
Change-Id: I587f5a69051f783753f5855d2f51242aa1cd5bc4
Signed-off-by: Delphine CC Chiu <Delphine_CC_Chiu@wiwynn.com>
---
 configurations/yosemite4_floatingfalls.json   | 19 ++++++++++++++-
 configurations/yosemite4_sentineldome_t1.json |  8 +++++++
 .../yosemite4_sentineldome_t1_retimer.json    |  8 +++++++
 configurations/yosemite4_sentineldome_t2.json |  8 +++++++
 .../yosemite4_sentineldome_t2_retimer.json    |  8 +++++++
 configurations/yosemite4_wailuafalls.json     | 24 +++++++++++++++++++
 schemas/legacy.json                           |  6 +++++
 7 files changed, 80 insertions(+), 1 deletion(-)

diff --git a/configurations/yosemite4_floatingfalls.json b/configurations/yosemite4_floatingfalls.json
index 1ee4e91..5a9cb31 100644
--- a/configurations/yosemite4_floatingfalls.json
+++ b/configurations/yosemite4_floatingfalls.json
@@ -1,5 +1,22 @@
 {
-    "Exposes": [],
+    "Exposes": [
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10 + 1",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10 + 3",
+            "Name": "CXL",
+            "Type": "MCTPEndpoint"
+        }
+    ],
     "Name": "Yosemite 4 Floating Falls Slot $bus % 15",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Floating Falls', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4'})",
     "Type": "Board",
diff --git a/configurations/yosemite4_sentineldome_t1.json b/configurations/yosemite4_sentineldome_t1.json
index 6c0096b..524d41b 100644
--- a/configurations/yosemite4_sentineldome_t1.json
+++ b/configurations/yosemite4_sentineldome_t1.json
@@ -338,6 +338,14 @@
             "Zones": [
                 "Zone 1"
             ]
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
         }
     ],
     "Name": "Yosemite 4 Sentinel Dome T1 Slot $bus % 15",
diff --git a/configurations/yosemite4_sentineldome_t1_retimer.json b/configurations/yosemite4_sentineldome_t1_retimer.json
index 6bf37b4..11a2bb4 100644
--- a/configurations/yosemite4_sentineldome_t1_retimer.json
+++ b/configurations/yosemite4_sentineldome_t1_retimer.json
@@ -398,6 +398,14 @@
             "Zones": [
                 "Zone 1"
             ]
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
         }
     ],
     "Name": "Yosemite 4 Sentinel Dome T1 with Retimer Slot $bus % 15",
diff --git a/configurations/yosemite4_sentineldome_t2.json b/configurations/yosemite4_sentineldome_t2.json
index 322e400..32814db 100644
--- a/configurations/yosemite4_sentineldome_t2.json
+++ b/configurations/yosemite4_sentineldome_t2.json
@@ -399,6 +399,14 @@
             "Zones": [
                 "Zone 1"
             ]
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
         }
     ],
     "Name": "Yosemite 4 Sentinel Dome T2 Slot $bus % 15",
diff --git a/configurations/yosemite4_sentineldome_t2_retimer.json b/configurations/yosemite4_sentineldome_t2_retimer.json
index 74695e7..f2c34e0 100644
--- a/configurations/yosemite4_sentineldome_t2_retimer.json
+++ b/configurations/yosemite4_sentineldome_t2_retimer.json
@@ -459,6 +459,14 @@
             "Zones": [
                 "Zone 1"
             ]
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
         }
     ],
     "Name": "Yosemite 4 Sentinel Dome T2 with Retimer Slot $bus % 15",
diff --git a/configurations/yosemite4_wailuafalls.json b/configurations/yosemite4_wailuafalls.json
index 9ba7fd9..f1b566a 100644
--- a/configurations/yosemite4_wailuafalls.json
+++ b/configurations/yosemite4_wailuafalls.json
@@ -214,6 +214,30 @@
             "Zones": [
                 "Zone 1"
             ]
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10 + 2",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10 + 4",
+            "Name": "CXL1",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": ["0x20"],
+            "Bus": "$bus % 16",
+            "Class": "I2C",
+            "EndpointId": "$bus % 15 * 10 + 5",
+            "Name": "CXL2",
+            "Type": "MCTPEndpoint"
         }
     ],
     "Name": "Yosemite 4 Wailua Falls Slot $bus % 15",
diff --git a/schemas/legacy.json b/schemas/legacy.json
index e56e8e9..17db8d1 100644
--- a/schemas/legacy.json
+++ b/schemas/legacy.json
@@ -68,6 +68,9 @@
                 "EntityInstance": {
                     "$ref": "#/$defs/Types/EntityInstance"
                 },
+                "EndpointId": {
+                    "$ref": "#/$defs/Types/EndpointId"
+                },
                 "FaultIndex": {
                     "$ref": "#/$defs/Types/FaultIndex"
                 },
@@ -871,6 +874,9 @@
             "EntityInstance": {
                 "type": "number"
             },
+            "EndpointId": {
+                "type": ["string", "number"]
+            },
             "FaultIndex": {
                 "type": "number"
             },
-- 
2.44.2

